src/popup/pages/SignMessage.vue

```diff
@@ -1,78 +1,41 @@
-<template>
-  <div class="sign-message">
-    <div class="section-title">
-      Sign message for
-    </div>
-
-    <div class="url-bar">
-      {{ callbackOrigin }}
-    </div>
-
-    <div class="section-title">
-      Message
-    </div>
-
-    <div class="tip-note-preview">
-      {{ $route.query.message }}
-    </div>
-
-    <BtnMain @click="sendAddress">
-      {{ $t('pages.tipPage.confirm') }}
-    </BtnMain>
-    <BtnMain @click="openCallbackOrGoHome(false)">
-      {{ $t('pages.tipPage.cancel') }}
-    </BtnMain>
-  </div>
-</template>
-
 <script lang="ts">
-import { defineComponent } from '@vue/composition-api';
+import { defineComponent, onMounted } from '@vue/composition-api';
 import { useDeepLinkApi, useSdk } from '../../composables';
-import BtnMain from '../components/buttons/BtnMain.vue';
+import { MODAL_MESSAGE_SIGN, handleUnknownError } from '../utils';
 
 export default defineComponent({
   name: 'SignMessage',
-  components: {
-    BtnMain,
-  },
   setup(props, { root }) {
-    const { openCallbackOrGoHome } = useDeepLinkApi({ router: root.$router });
-    const { getSdk } = useSdk({ store: root.$store });
-
-    const sendAddress = async () => {
-      const sdk = await getSdk();
-      const signature = await sdk.signMessage(root.$route.query.message);
-      const signatureHex = Buffer.from(signature).toString('hex');
-      openCallbackOrGoHome(true, { signature: signatureHex });
-    };
-
-    return {
-      sendAddress,
-    };
+    onMounted(async () => {
+      const { callbackOrigin, openCallbackOrGoHome } = useDeepLinkApi({ router: root.$router });
+      const { getSdk } = useSdk({ store: root.$store });
+
+      try {
+        const sdk = await getSdk();
+        const { message } = root.$route.query;
+
+        await root.$store.dispatch('modals/open', {
+          name: MODAL_MESSAGE_SIGN,
+          message,
+          app: {
+            name: callbackOrigin.value?.host,
+            host: callbackOrigin.value?.host,
+            url: callbackOrigin.value?.origin,
+          },
+        });
+
+        const signature = await sdk.signMessage(message);
+        const signatureHex = Buffer.from(signature).toString('hex');
+        openCallbackOrGoHome(true, { signature: signatureHex });
+      } catch (e: any) {
+        openCallbackOrGoHome(false);
+
+        if (e.message !== 'Rejected by user') handleUnknownError(e);
```

Comment:
This comparison with `Rejected by user` or throwing error with this message is quite common in the system. I suggest to create some kind of abstraction layer for this.
For example by creating custom error:
```
class RejectedByUserError extends Error { ... }
```
and catching it this way:
```
catch(err) {
  if (err instanceof RejectedByUserError) {
    handleUnknownError(e);
  }
}
```